flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/control/PidController.java

```diff
@@ -173,32 +143,34 @@ public void setOutputLimit(double outputLimit){
      * Calculates the output to the system to compensate for the error.
      *
      * @param processVariable the process variable of the system.
-     * @param setPoint the desired set point.
+     * @param setpoint the desired set point.
      *
      * @return the compensation value from the PID loop calculation
      */
-    public double calculate(double processVariable, double setPoint){
+    public double calculate(double processVariable, double setpoint, double currentTimeSeconds) {
```

Comment:
You could receive a `Clock` object in the constructor instead of receiving time in here. Will make it easier for users.
Plus you could make an overload which uses the default clock:
```Java
public PidController(DoubleSupplier kp....., Clock clock) {
    ....
}
public PidController(DoubleSupplier kp.....) {
    this(kp, ......, RunningRobot.getInstance().getClock());
}
```

Reply:
No. I dangerous because the clock won't be synced with the values, so I use a clock where I supply the process variable myself.

Reply:
Explain?

Reply:
yeah